apps/backend/src/modules/contributors/contributor.repository.ts

```diff
@@ -103,8 +104,13 @@ class ContributorRepository implements Repository {
 			)
 			.leftJoin("git_emails", "contributors.id", "git_emails.contributor_id")
 			.leftJoin("activity_logs", "git_emails.id", "activity_logs.git_email_id")
-			.leftJoin("projects", "activity_logs.project_id", "projects.id")
-			.where("projects.id", projectId)
+			.leftJoin("projects", "activity_logs.project_id", "projects.id");
+
+		if (!hasRootPermission) {
+			query.whereIn("projects.id", projectIds);
+		}
```

Comment:
Same as my prev comment, looks like a business logic. But again, I'm not sure, better to clarify it with coaches.